generators/client/templates/react/src/main/webapp/app/modules/administration/audits/audits.tsx

```diff
@@ -1,65 +1,146 @@
 import * as React from 'react';
 import { connect } from 'react-redux';
-
-import { Table } from 'reactstrap';
+import { Input } from 'reactstrap';
+import {
+  Translate,
+  ICrudGetAction,
+  TextFormat,
+  JhiPagination,
+  getPaginationItemsNumber,
+  getSortState,
+  IPaginationState
+} from 'react-jhipster';
+import { FaSort } from 'react-icons/lib/fa';
 
 import { getAudits } from '../../../reducers/administration';
+import { APP_TIMESTAMP_FORMAT } from '../../../config/constants';
+import { ITEMS_PER_PAGE } from '../../../shared/util/pagination.constants';
 
 export interface IAuditsPageProps {
   isFetching?: boolean;
   audits: any[];
   getAudits: Function;
+  totalItems: 0;
+  history: any;
+  location: any;
+  onChangeFromDate: any;
+  onChangeToDate: any;
 }
 
-export class AuditsPage extends React.Component<IAuditsPageProps, undefined> {
+export class AuditsPage extends React.Component<IAuditsPageProps, IPaginationState> {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      ...getSortState(props.location, ITEMS_PER_PAGE),
+      fromDate: this.previousMonth(),
+      toDate: this.today()
+    };
+  }
 
   componentDidMount() {
-    this.props.getAudits();
+    this.getAudits();
+  }
+
+  onChangeFromDate = evt => {
+    this.setState({
+      fromDate: evt.target.value
+    }, () => this.getAudits());
+  }
+  onChangeToDate = evt => {
+    this.setState({
+      toDate: evt.target.value
+    }, () => this.getAudits());
+  }
+
+  previousMonth(): string {
+    const now: Date = new Date();
+    const fromDate = now.getMonth() === 0
+      ? new Date(now.getFullYear() - 1, 11, now.getDate())
+      : new Date(now.getFullYear(), now.getMonth() - 1, now.getDate());
+    return fromDate.toISOString().slice(0, 10);
+  }
+
+  today(): string {
+    // Today + 1 day - needed if the current day must be included
+    const today: Date = new Date();
+    today.setDate(today.getDate() + 1);
+    const toDate = new Date(today.getFullYear(), today.getMonth(), today.getDate());
+    return toDate.toISOString().slice(0, 10);
+  }
+
+  sort = prop => () => {
+    this.setState({
+      order: this.state.order === 'asc' ? 'desc' : 'asc',
+      sort: prop
+    }, () => this.transition());
   }
 
-  getAuditList = () => {
-    if (!this.props.isFetching) {
-      this.props.getAudits();
-    }
+  transition() {
```

Comment:
use const arrow method so that there is no bind issues.
`transition = () => {}`